### PR TITLE
refactor: update insert_rows.py for more efficient streaming

### LINKED ISSUE #211: Improve guidance for streaming inserts
Suggesting we update the insert_rows.py code to showcase how to explicitly define a schema to avoid excessive calls to get_table().

### PR DESCRIPTION
**Pull request by jmgreger**
Added in the use of selected_fields=schema to explicitly define the schema. This prevents excessive calling of table metadata when streaming large volumes of data.

There may be a better way to showcase this in the code, but I wanted to put this out there for review.

Fixes #211  🦕

### CODE REVIEW

**Comment by jmgreger** [comment]
> Please ~rebase to remove the release-related changes from this PR (the first three commits).~ Oops, my bad: the issue is that your PR targets the `release-v1.27.0` branch, instead of `master`.

Ahh, sorry about that. I don't submit code very often :/.  Should I make a new PR against master instead?

**Comment by jmgreger** [comment]
I re-based this off of the master branch, now it just needs the files added by the chore bot to be removed. Doesn't look like I have rights to do that though.

**Comment by tseaver** [comment]
@jmgreger If you use `git rebase -i` in your local clone, you should be able to delete the bot-generated commits.  E.g.:

```bash
$ git checkout master
$ git fetch --all upstream  #  assuming this repo's remote is named 'upstream'.
$ git pull upstream/master
$ git checkout patch-1
$ git rebase -i master  # delete first three commits and the last merge commit
$ git push -f origin patch-1
```

As an alternative, you could just cherry-pick the patch from [this commit](https://github.com/googleapis/python-bigquery/pull/212/commits/471518a3b9d6da36167528f035e9a486984b1503) into a new branch against current master, and then submit a new PR from that branch.

File: samples/table_insert_rows.py [hunk 1]
```diff
@@ -12,23 +12,28 @@
...
+# [START bigquery_table_insert_rows]
+
+from google.cloud import bigquery
+
+# Construct a BigQuery client object.
+client = bigquery.Client()
+
+# TODO(developer): Set table_id to the ID of the model to fetch.
+table_id = "your-project.your_dataset.your_table"
+#Explicitly define the schema. (Optional)
```
Comment (tswast) [inline samples/table_insert_rows.py:24]: Start comments with a space. Also, I think this warrants more explanation.

```suggestion
# The client converts Python objects to JSON-serialization, but this requires a schema.
# A schema can be fetched by calling `client.get_table`. If inserting many rows, it is
# suggested that you cache the schema. The backend API for `client.insert_rows` supports
# a much higher QPS than the backend API for `client.get_table`.
```

File: samples/table_insert_rows.py [hunk 2]
```diff
@@ -12,23 +12,28 @@
...
+    ]
+rows_to_insert = [(u"Phred Phlyntstone", 32), (u"Wylma Phlyntstone", 29)] #populate data for entry
+
+try:
+    if 'schema' not in globals():
+        schema = client.get_table(table_id).schema  # Get table schema if none provided
+    client.insert_rows(table_id, selected_fields=schema, rows_to_insert)  # Stream data to BQ
+    print("New rows have been added.")
+except ValueError:
+    print("Table’s schema is not set or rows is not a Sequence.")
```
Comment (tswast) [inline samples/table_insert_rows.py:37]: No need to catch this `ValueError`. Just let the exception bubble up.

**Comment by leahecole** [comment]
LGTM once Tim's comments are addressed and as long as tests pass

**Comment by plamut** [comment]
Closing this in favor of #253 as per comment in that PR.

**Pull request closed without merging**